apps/extension/src/routes/popup/approval/transaction/index.tsx

```diff
@@ -51,6 +51,17 @@ export const TransactionApproval = () => {
   return (
     <div className='flex h-screen flex-col'>
       <div className='flex grow flex-col overflow-auto p-[30px] pt-10'>
+        {selectedTransactionViewName === TransactionViewTab.SENDER && symbol !== 'UM' && (
+          <div
+            style={{ marginBottom: '16px' }}
+            className='rounded border border-yellow-500 p-2 text-yellow-500 text-sm'
+          >
+            <span className='block text-center font-bold'>⚠ Privacy Warning:</span>
+            Transaction uses a non-native fee token. To reduce gas costs and protect your privacy,
+            maintain an UM balance for fees.
+          </div>
+        )}
+
```

Comment:
alternatively, we can implement a tooltip with a hover animation

<img width="512" alt="Screenshot 2024-11-25 at 12 31 47 AM" src="https://github.com/user-attachments/assets/df71a437-8cdb-49b0-bb26-90fafadfc3ba">

Reply:
I think it's fine, hopefully the user only sees this once or twice